### ISSUE TITLE
AttributeError: module 'torch.backends' has no attribute 'mps'

### ISSUE THREAD

**Issue by ravijo** [pytorch/examples#1134]
## Context

* Pytorch version: 1.9.0+cu111
* Operating System and version: Ubuntu 18.04 LTS

## Your Environment

* Installed using source? [yes/no]: No (installed using `pip`)
* Are you planning to deploy it using docker container? [yes/no]: No
* Is it a CPU or GPU environment?: GPU
* Which example are you using: imagenet
* Link to code or data to repro [if any]: [pytorch/examples/imagenet/main.py](https://github.com/pytorch/examples/blob/main/imagenet/main.py)

## Expected Behavior

There should be no exception.

## Current Behavior

```
$ python3 main.py -a resnet18 --dummy
=> creating model 'resnet18'
Traceback (most recent call last):
  File "main.py", line 511, in <module>
    main()
  File "main.py", line 120, in main
    main_worker(args.gpu, ngpus_per_node, args)
  File "main.py", line 171, in main_worker
    elif torch.backends.mps.is_available():
AttributeError: module 'torch.backends' has no attribute 'mps'
```

## Possible Solution

Set GPU ID to 0. This can be done at the default settings show below:
https://github.com/pytorch/examples/pull/1135/commits/4bc313ea7c7287d8f6d9f7dfb5cbf3a192a4c6bc

Created a pull request: https://github.com/pytorch/examples/pull/1135

**Comment by Ansor-ZJJ** [comment]
I also encountered the same problem. I annotated the 'mps' code and then set to use multiple Gpus. "model = torch.nn.DataParallel allel(model). But I don't know what's the difference between this and single-node multi-GPU training mode


```
    if not torch.cuda.is_available() and not torch.backends.mps.is_available():
        print('using CPU, this will be slow')
    elif args.distributed:
        # For multiprocessing distributed, DistributedDataParallel constructor
        # should always set the single device scope, otherwise,
        # DistributedDataParallel will use all available devices.
        if torch.cuda.is_available():
            if args.gpu is not None:
                torch.cuda.set_device(args.gpu)
                model.cuda(args.gpu)
                # When using a single GPU per process and per
                # DistributedDataParallel, we need to divide the batch size
                # ourselves based on the total number of GPUs of the current node.
                args.batch_size = int(args.batch_size / ngpus_per_node)
                args.workers = int((args.workers + ngpus_per_node - 1) / ngpus_per_node)
                model = torch.nn.parallel.DistributedDataParallel(model, device_ids=[args.gpu])
            else:
                model.cuda()
                # DistributedDataParallel will divide and allocate batch_size to all
                # available GPUs if device_ids are not set
                model = torch.nn.parallel.DistributedDataParallel(model)
    elif args.gpu is not None and torch.cuda.is_available():
        torch.cuda.set_device(args.gpu)
        model = model.cuda(args.gpu)
    # elif torch.backends.mps.is_available():
    #     device = torch.device("mps")
    #     model = model.to(device)
    else:
        # DataParallel will divide and allocate batch_size to all available GPUs
        if args.arch.startswith('alexnet') or args.arch.startswith('vgg'):
            model.features = torch.nn.DataParallel(model.features)
            model.cuda()
        else:
            model = torch.nn.DataParallel(model).cuda()


    if torch.cuda.is_available():
        if args.gpu:
            device = torch.device('cuda:{}'.format(args.gpu))
        else:
            device = torch.device("cuda")
    # elif torch.backends.mps.is_available():
    #     device = torch.device("mps")
    else:
        device = torch.device("cpu")
    # define loss function (criterion), optimizer, and learning rate scheduler

```

**Comment by msaroufim** [comment]
@ravijo your PyTorch version is quite old, might be best to upgrade

**Comment by rmib200** [comment]
I am having the same issue but I am on Windows. My Torch version is 2.1.2

**Comment by ravijo** [comment]
@rmib200 
I forgot the details already, but can you please upgrade your PyTorch and check it please?

**Comment by bowen480** [comment]
> @ravijo your PyTorch version is quite old, might be best to upgrade

do you solve this problem?

**Comment by ravijo** [comment]
@bowen480 
Yes, I upgraded PyTorch to get rid of this error.

**Comment by 287400117** [comment]
> @bowen480 Yes, I upgraded PyTorch to get rid of this error.

I am having the same issue，which PyTorch version you upgraded?

**Comment by ravijo** [comment]
@287400117 It's been over a year, and I unfortunately don't remember the PyTorch version. My apologies

**Comment by 287400117** [comment]
> @287400117 It's been over a year, and I unfortunately don't remember the PyTorch version. My apologies
thank you for replay，My issue was resolved after upgrading to pytorch 1.13.1

**Comment by 287400117** [comment]
> @287400117 It's been over a year, and I unfortunately don't remember the PyTorch version. My apologies

thank you for replay，My issue was resolved after upgrading to pytorch 1.13.1
